ynr/apps/bulk_adding/helpers.py

```diff
@@ -0,0 +1,93 @@
+from popolo.models import Membership, Person
+
+from candidates.models import (LoggedAction, MembershipExtra, PersonExtra,
+                               raise_if_unsafe_to_delete)
+from candidates.models.auth import check_creation_allowed
+from candidates.views.version_data import get_change_metadata, get_client_ip
+
+
+def add_person(request, person_data):
```

Comment:
I'm confused about this method. On the face of it, it looks like you're missing a call to `person.save()`. I checked to see if there is an override on `person_extra.save()` so it also saves `person` but it doesn't seem like it does.. this one might be my lack of familiarity with the codebase as opposed to oversight.

Reply:
The line directly after this comment:

`person = Person.objects.create(name=person_data['name'])`

From the docs:
[`create(**kwargs)`: A convenience method for creating an object and saving it all in one step](https://docs.djangoproject.com/en/1.9/ref/models/querysets/#django.db.models.query.QuerySet.create)

Reply:
right, yep.. lack of reading the code properly rather than lack of familiarity with the codebase :D